sql/core/src/main/scala/org/apache/spark/sql/jdbc/PostgresDialect.scala

```diff
@@ -70,21 +70,21 @@ private case class PostgresDialect() extends JdbcDialect with SQLConfHelper {
       case Types.OTHER => Some(StringType)
       case _ if "text".equalsIgnoreCase(typeName) => Some(StringType) // sqlType is Types.VARCHAR
       case Types.ARRAY =>
-        val scale = md.build().getLong("scale").toInt
-        val isTimestampNTZ = md.build().getBoolean("isTimestampNTZ")
         // postgres array type names start with underscore
-        toCatalystType(typeName.drop(1), size, scale, isTimestampNTZ).map(ArrayType(_))
+        toCatalystType(typeName.drop(1), size, md).map(ArrayType(_))
       case _ => None
     }
   }
 
   private def toCatalystType(
       typeName: String,
       precision: Int,
-      scale: Int,
-      isTimestampNTZ: Boolean): Option[DataType] = typeName match {
+      md: MetadataBuilder): Option[DataType] = typeName match {
     case "bool" => Some(BooleanType)
-    case "bit" => Some(BinaryType)
+    case "bit" if precision == 1 => Some(BooleanType)
```

Comment:
is it possible to keep the previous type mapping and only fix the read code? It seems a bit weird to me to have different type mapping based on the bit length.

Reply:
- This aligns atomic bit(1), see https://github.com/apache/spark/pull/45751/files#diff-374d2da08d88160979e0fedb2ded14e5fa8ed04942a94cdd9672775332642226L55
- pgjdbc has a boolean array decoder for bit(1)[], while it also wrongly tries to decode bit(n)[] anyway.

Reply:
Do you have any other concerns, @cloud-fan ?

Reply:
> This aligns atomic bit(1), see ...

I see, internal consistency is important.

Reply:
Thank you for the double-check and post-review. @cloud-fan